mlflow/tracing/python_api.py

```diff
@@ -122,7 +127,58 @@ def start_span(
             mlflow_span = NoOpMLflowSpanWrapper()
             yield mlflow_span
     finally:
-        mlflow_span._end()
+        mlflow_span.end()
+
+
+def start_detached_span(
```

Comment:
This is a great approach. One nit on the naming, though. 
What are your thoughts on `start_manual_span` to indicate that the span needs to be handled explicitly for closing the tracer context?